src/notifcation/NotificationQueueingService.java

```diff
@@ -0,0 +1,76 @@
+package notifcation;
+
+import hospital.Patient;
+import hospital.PatientStatus;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ *
+ * @author maximdumont
+ * 
+ * Notifcation Service to be used to send messages to a sepcific output system
+ */
+public class NotificationQueueingService {
+    
+    //Acts as the master queue to be used to keep track of patients and their changes
+    private Map<String, Patient> patientsMap;
+
+    public NotificationQueueingService() {
+        this.patientsMap = new HashMap<>();
+    }
+
+    //subscribe a group of patients to be notified when changes occur
+    public void subscribeAll(Patient[] patients) {
+        for (Patient p : patients) {
+            subscribe(p);
+        }
+    }
+
+    //subscribe single patient to changes
+    public void subscribe(Patient patient) {
+        patientsMap.putIfAbsent(patient.getPatientId(), patient);
+    }
+
+    //modify patient status in existing patient based on identifier
+    public void change(String patientIdentifier, PatientStatus status) {
```

Comment:
I'm not sure we need this in the notification queue. The update status controller will set the patient status so this is a bit redundant; the status will already have been set in the referenced Patient object. How do you see it being used?